File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IChannelCellParsingState.cs
//*******************************************************************************************************
//  IChannelCellParsingState.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/18/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace PCS.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of the parsing state of any kind of frame cell.
    /// </summary>
    public interface IChannelCellParsingState : IChannelParsingState
    {
        /// <summary>
        /// Gets or sets the number of phasor elements associated with the <see cref="IChannelCell"/>.
        /// </summary>
        int PhasorCount { get; set; }

        /// <summary>
        /// Gets or sets the number of analog elements associated with the <see cref="IChannelCell"/>.
        /// </summary>
        int AnalogCount { get; set; }

        /// <summary>
        /// Gets or sets the number of digital elements associated with the <see cref="IChannelCell"/>.
        /// </summary>
        int DigitalCount { get; set; }
    }
}

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols (VB)/IChannelCellParsingState.vb
'*******************************************************************************************************
'  IChannelCellParsingState.vb - Channel data cell parsing state interface
'  Copyright © 2008 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2008
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/18/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

''' <summary>This interface represents a protocol independent parsing state of any kind of data cell.</summary>
Public Interface IChannelCellParsingState

    Inherits IChannelParsingState

    Property PhasorCount() As Int32

    Property AnalogCount() As Int32

    Property DigitalCount() As Int32

End Interface